source/decode.c

```diff
@@ -463,15 +567,27 @@ static int s_linestate_header(struct aws_http_decoder *decoder, struct aws_byte_
 
 static int s_linestate_request(struct aws_http_decoder *decoder, struct aws_byte_cursor input) {
     struct aws_byte_cursor cursors[3];
-    if (s_byte_buf_split(input, cursors, ' ', 3) != 3) {
+    int err = s_cursor_split_exactly_n_times(input, ' ', cursors, 3); /* extra spaces not allowed */
+    if (err) {
+        AWS_LOGF_ERROR(
+            AWS_LS_HTTP_STREAM, "id=%p: Incoming request line has wrong number of spaces.", decoder->logging_id);
         return aws_raise_error(AWS_ERROR_HTTP_PARSE);
     }
+
+    for (size_t i = 0; i < AWS_ARRAY_SIZE(cursors); ++i) {
+        if (cursors[i].len == 0) {
+            AWS_LOGF_ERROR(AWS_LS_HTTP_STREAM, "id=%p: Incoming request line has empty values.", decoder->logging_id);
+        }
+    }
+
     struct aws_byte_cursor method = cursors[0];
     struct aws_byte_cursor uri = cursors[1];
     struct aws_byte_cursor version = cursors[2];
 
     struct aws_byte_cursor version_expected = aws_http_version_to_str(AWS_HTTP_VERSION_1_1);
     if (!aws_byte_cursor_eq(&version, &version_expected)) {
+        AWS_LOGF_ERROR(
+            AWS_LS_HTTP_STREAM, "id=%p: Incoming request uses unsupported HTTP version.", decoder->logging_id);
```

Comment:
Why not log the unexpected version?

Reply:
I was on the fence about whether to log invalid stuff. Like, if a header is invalid, do I log the bad string? Is this any kind of security risk? Or might I spit out 1000 lines of binary data?

thoughts?

Reply:
logging it.
It looks lovely btw